aten/src/ATen/UndefinedTensor.h

```diff
@@ -7,6 +7,7 @@ namespace at {
 struct AT_API UndefinedTensor final : public TensorImpl {
 public:
   static inline UndefinedTensor * singleton() {
+    static UndefinedTensor _singleton;
```

Comment:
This lazily initializes `_singleton`, which we don't want.

Reply:
Why?

Reply:
According to @gchanan, this is because a lazy static initializer means that the compiler must emit a conditional test to see if the variable is initialized or not.

Evidence: https://godbolt.org/g/a3NViC

Reply:
Is that really such a big deal? It's not even an atomic instruction (1B `mov`), so I doubt it has _any_ impact on actual perf.